onnx/mapping.py

```diff
@@ -1,53 +1,51 @@
 # SPDX-License-Identifier: Apache-2.0
 
-from onnx import TensorProto, SequenceProto, OptionalProto
+import warnings
+from typing import Any
 import numpy as np  # type: ignore
+from onnx import TensorProto, SequenceProto, OptionalProto
 
-# This map is used for converting TensorProto values into Numpy arrays
-TENSOR_TYPE_TO_NP_TYPE = {
-    int(TensorProto.FLOAT): np.dtype('float32'),
-    int(TensorProto.UINT8): np.dtype('uint8'),
-    int(TensorProto.INT8): np.dtype('int8'),
-    int(TensorProto.UINT16): np.dtype('uint16'),
-    int(TensorProto.INT16): np.dtype('int16'),
-    int(TensorProto.INT32): np.dtype('int32'),
-    int(TensorProto.INT64): np.dtype('int64'),
-    int(TensorProto.BOOL): np.dtype('bool'),
-    int(TensorProto.FLOAT16): np.dtype('float16'),
-    int(TensorProto.BFLOAT16): np.dtype('float32'),  # Native numpy does not support bfloat16 so now use float32 for bf16 values
-    int(TensorProto.DOUBLE): np.dtype('float64'),
-    int(TensorProto.COMPLEX64): np.dtype('complex64'),
-    int(TensorProto.COMPLEX128): np.dtype('complex128'),
-    int(TensorProto.UINT32): np.dtype('uint32'),
-    int(TensorProto.UINT64): np.dtype('uint64'),
-    int(TensorProto.STRING): np.dtype('object')
+# tensor_dtype: (numpy type, storage type, string name)
+TENSOR_TYPE_MAP = {
+    int(TensorProto.FLOAT): (np.dtype('float32'), int(TensorProto.FLOAT), 'TensorProto.FLOAT'),
+    int(TensorProto.UINT8): (np.dtype('uint8'), int(TensorProto.INT32), 'TensorProto.UINT8'),
+    int(TensorProto.INT8): (np.dtype('int8'), int(TensorProto.INT32), 'TensorProto.INT8'),
+    int(TensorProto.UINT16): (np.dtype('uint16'), int(TensorProto.INT32), 'TensorProto.UINT16'),
+    int(TensorProto.INT16): (np.dtype('int16'), int(TensorProto.INT32), 'TensorProto.INT16'),
+    int(TensorProto.INT32): (np.dtype('int32'), int(TensorProto.INT32), 'TensorProto.INT32'),
+    int(TensorProto.INT64): (np.dtype('int64'), int(TensorProto.INT64), 'TensorProto.INT64'),
+    int(TensorProto.BOOL): (np.dtype('bool'), int(TensorProto.INT32), 'TensorProto.BOOL'),
+    int(TensorProto.FLOAT16): (np.dtype('float16'), int(TensorProto.UINT16), 'TensorProto.FLOAT16'),
```

Comment:
not sure about the value here

Reply:
I merged previous `TENSOR_TYPE_TO_NP_TYPE` and `TENSOR_TYPE_TO_STORAGE_TENSOR_TYPE` with the new naming map into the single map `TENSOR_TYPE_MAP`, because they have the same set of keys (TensorProto data_type). They were the ground truth here and not tested previously either. If the unit test is really needed, I am inclined to have another PR for it since this PR is already a big one and actually I didn't change the original content here.